src/Authenticator/LoginUrlCheckerInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Authenticator;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * LoginUrlCheckerInterface
+ */
+interface LoginUrlCheckerInterface
```

Comment:
Do we need an interface for an internal class? There's a very little chance that somebody would replace that checker with their own implementation. Wouldn't just a callable/invokable class be sufficent?

Reply:
Yes, there should be an interface IMHO. I haven't yet added a type check on this and actually, I might use my own checker. The interface provides a clear implementation. The invokeable stuff feels so javascriptish as well. I don't like the in-transparency of them.

I haven't yet added a type check that will check for the interface intentionally, I hope we might get some kind of more abstract DI than the locators that are always provided via traits. If / when this happens we want to have an interface.

I've added the trait back, but it doesn't do any logic now, it's just there to share the logic (DRY) between the two authenticators and acts as "glue" between the object and the classes, it's basically a getter/locator. I've also refactored some code in FormAuthenticator::authenticate() :)

Reply:
Right, on the second thought the interface is OK. You could implement the cake router checker with that :+1:

Reply:
I can do that later, but before doing more work here, can we just merge this PR? 😄 

I need to leave for some time and take care of my horse any way.

Reply:
I have just one more concern regarding configuring the checker. Since different checkers might require different config, maybe we should extract this into an array and build the checker with this config?

Reply:
I don't think the constructor belongs into the interface, you can already pass the config as third arg. Nor would I create a dependency on the ConfigTrait.

Reply:
OK, so maybe let's wrap the config into an array and pass as the second argument to the checker.

Since the checker is another object, I don't like how it's config is mixed with authenticator config.

I have something like this in mind:
```php
$service->loadAuthenticator('Form', [
    'loginUrl' => '/some-url',
    'loginUrlChecker' => [
        'className' => CustomChecker::class,
        'customOption' => true
    ]
]);

//internally:
$checker->check('/some-url', [
    'customOption' => true
]);

Reply:
I'm OK with passing the array, I don't like yet another locator. Also can we keep the check() taking the request as first arg, so in total the three we already have? Otherwise you would have to put the constructor in the interface.

Reply:
Current API is ok. I just want to have the ability to pass custom options to the checker. Config array seems like a good solution.

Sorry for the wrong example, of course there should be the request as a first argument.

Reply:
@robertpustulka no problem. I've changed the config but I'm not happy with it either. A 2nd level array config opens always room for undefined index errors.

Reply:
You can merge options with defaults to ensure that indexes are present

Reply:
Which means adding the config trait again. :( Feel free to change the code, I'm leaving again, now for my Wing Chun training.

Reply:
You don’t need that.

Just `$options += $defaults` should do the job.

Reply:
I'll add that now and we can merge it after that is done?